services/scheduler/src/main/java/org/mule/service/scheduler/internal/DefaultSchedulerService.java

```diff
@@ -221,6 +235,14 @@ public Scheduler customScheduler(SchedulerConfig config, int queueSize) {
     return customScheduler;
   }
 
+  private ThreadGroup resolveThreadGroupForCustomScheduler(SchedulerConfig config) {
+    if (config.getWaitDispatchingToBusyScheduler() != null && config.getWaitDispatchingToBusyScheduler()) {
```

Comment:
if i reverse, the case where the value is null with throw a null pointer exception. It is validated in `DefaultSchedulerServiceTestCase`